nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -736,7 +736,7 @@ public static Object convertRecordFieldtoObject(final Object value, final DataTy
     }
 
     public static boolean isMapTypeCompatible(final Object value) {
-        return value != null && value instanceof Map;
+        return value != null && (value instanceof Map || value instanceof MapRecord);
```

Comment:
This feels a little hacky to me, but the alternative was to refactor a bunch of code that validates record fields, so we could instead validate map entries. It would also entail changing the JSON Reader(s) (at the least) to generate Maps instead of MapRecords as it does now. So I'm checking specifically for MapRecord (not Record) to say that it's compatible with a MAP type, which is consistent with the example schema and data (a MapRecord is created but the type is MAP<STRING>).